recipes-graphics/xorg-xserver/xserver-xf86-config/xenclient-uivm/xorg.conf

```diff
@@ -23,7 +23,7 @@ EndSection
 Section "InputDevice"
         Identifier "Keyboard0"
         Driver     "evdev"
-        Option     "Device"    "/dev/input/event0"
+        Option     "Device"    "/dev/input/event2"
```

Comment:
Interesting, why is that?
On Xen 4.10.0 `/proc/bus/input/devices` does list `xenbus/device/vkbd/1` for `input{0,1}` instead of `xenbus/device/vkbd/0` on Xen 4.9.2.

Reply:
Also is there a reason Xorg can't automatically pick up the input device?

Reply:
I don't know

Reply:
I dropped the 2 mouse and 1 keyboard  InputDevice sections and ServerLayout section and UIVM came up with mouse and keyboard working.  The mouse speed may be different? It's using /dev/input/mice for the "default pointer".  Xorg.0.log shows X detecting and skipping devices /dev/input/mouse[01] and /dev/input/event[0-3] because ServerFlags has ```Option "AutoAddDevices" "false"```

The X log also shows two keyboard devices: /dev/input/event0 and /dev/input/event2

Reply:
This is a slippery slope... We expose a relative mouse, an absolute mouse and a keyboard.
Changing this should be a separate effort, that will require a lot of testing with a lot of hardware.
Here I'm just adapting the existing solution to Xen 4.10.

Reply:
Yes, I agree this is outside the scope of the Xen upgrade.

Reply:
Makes sense, see [OXT-1320](https://openxt.atlassian.net/browse/OXT-1320).

Reply:
@jean-edouard, was this xorg.conf change needed because there was no keyboard input?